File: src/NoteKeeperMVC/NoteKeeperMVC/Scripts/komvvm/viewModels/NoteViewModel.js
(function (nk, $) {
    // =====================================================
    // =========== **** Main Note View Model **** =========
    // This is the simple note view model used to display notes
    // on a board.
    // =====================================================
    
    // Checking code that creates the proppert namespace under the global namespace
    var viewModels = nk.ViewModels || {};
    nk.ViewModels = viewModels;

    nk.ViewModels.NoteViewModel = function (id, boardId, title, text, top, left) {
        var self = this;

        // The identifier observables for each note.
        self.id = id;
        self.boardId = boardId;

        // Textual observables for each note.
        self.title = ko.observable(title);
        self.text = ko.observable(text);
        
        // The editable textual observables
        self.editingText = ko.observable(text);
        self.editingTitle = ko.observable(title);
        
        // Positional observables for each note
        self.top = ko.observable(top);
        self.left = ko.observable(left);
    };
})(window.NK = window.NK || {}, jQuery);

File: src/NoteKeeperMVC/NoteKeeperMVC/ViewModels/Data/NoteViewModel.cs
using System;

namespace NK.Web.ViewModels.Data
{
    /// <summary>
    /// The View model used to display Note entity data on the client side (UI).
    /// </summary>
    public class NoteViewModel
    {
        #region Properties

        public Guid Id { get; set; }

        public Guid BoardId { get; set; }

        public string Title { get; set; }

        public string Text { get; set; }

        public string Top { get; set; }

        public string Left { get; set; }

        #endregion
    }
}